### PR TITLE
MGMT-4733 provide host in Operator requirements API

### PR DESCRIPTION
**Pull request by jakub-dzon**
Interface change is required to allow the OLM operator plugins to access inventory information while calculating host requirements.

### CODE REVIEW

File: internal/operators/cnv/cnv_operator.go [hunk 1]
```diff
@@ -128,8 +128,9 @@ func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
 	return &Operator
 }
 
-// GetHostRequirementsForRole provides operator's requirements towards host in a given role
-func (o *operator) GetHostRequirementsForRole(_ context.Context, _ *common.Cluster, role models.HostRole) (*models.ClusterHostRequirementsDetails, error) {
+// GetHostRequirements provides operator's requirements towards the host
+func (o *operator) GetHostRequirements(_ context.Context, _ *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	role := host.Role
 	switch role {
```
Comment (flaper87) [inline internal/operators/cnv/cnv_operator.go:134]: nitpick :smile: 

```suggestion
	switch host.Role {
```

**Review by flaper87 — APPROVED** [review]
/lgtm
/approve

File: internal/operators/cnv/cnv_operator.go (hunk 1 above)
Comment (jakub-dzon) [reply internal/operators/cnv/cnv_operator.go:134]: That was deliberate, as the role is used in one more place. I don't feel especially attached to that style, though:)

**Pull request merged**